frontend/device/android/device.lua

```diff
@@ -347,12 +347,27 @@ end
 -- to-do: implement fullscreen toggle in API19+
 local function canToggleFullscreen()
     local api = android.app.activity.sdkVersion
-    return api < 19, api
+    return true, api
 end
 
 -- toggle fullscreen API 19+
 function Device:_toggleFullscreenImmersive()
-    logger.dbg("ignoring fullscreen toggle, reason: always in immersive mode")
+
+    local width = android.getScreenWidth()
+    local height = android.getScreenHeight()
+    -- NOTE: Since we don't do HW rotation here, this should always match width
+    local available_width = android.getScreenAvailableWidth()
+    local available_height = android.getScreenAvailableHeight()
+
+    local is_fullscreen = android.isFullscreen()
+    logger.dbg(string.format("Change fullscreen to %s in _toggleFullscreenImmersive", not is_fullscreen))
+    android.setFullscreen(not is_fullscreen)
+    self.fullscreen = android.isFullscreen()
+    if self.fullscreen then
+        self:setViewport(0, 0, width, height)
+    else
+        self:setViewport(0, 0, available_width, available_height)
```

Comment:
is the viewport correct here?. If notifications are placed on top I would assume some offset on x/y based on current hw rotation mode.

Reply:
@pazos we are showing only navbar, so no need to calculate top offset.

Reply:
Sorry. Forgot about this PR :p. 

Yay. Indeed I got the intention of this PR completely wrong. It should work fine as is for navigation bar. I was thinking about the notifications above the screen.

It might produce some undersired behaviour on onyx devices which place the navigation bar on top but that's another issue and doesn't need to be addressed here.